Classes/Hook/ProcessDatamap/ProductInheritanceProcessDatamap.php

```diff
@@ -194,17 +236,54 @@ protected function fetchChildRecordIdentifiers($identifier): array
             return $queryBuilder
                 ->select('uid')
                 ->from(ProductRepository::TABLE_NAME)
-                ->where($queryBuilder->expr()->eq(
-                    'parent',
-                    $queryBuilder->createNamedParameter($identifier)
-                ))
+                ->where(
+                    $queryBuilder->expr()->eq(
+                        'parent',
+                        $queryBuilder->createNamedParameter($identifier)
+                    ),
+                    $queryBuilder->expr()->eq(
+                        'sys_language_uid',
+                        $queryBuilder->createNamedParameter($language, \PDO::PARAM_INT)
+                    )
+                )
                 ->execute()
                 ->fetchAll(FetchMode::COLUMN, 0);
         }
 
         return [];
     }
 
+    /**
+     * Calculate what product uid to use when calculating parent hash.
+     *
+     * @param bool $recordIsChild
+     * @param bool $recordIsParent
+     * @param int $parentProductId
+     * @param int $productId
+     * @param int $language
+     * @return int
+     */
+    protected function calculateParentHashProductId(
```

Comment:
Is this a calculation? Could also be a "find" or "findCorrect". Not exactly sure.

```suggestion
    protected function findCorrectParentHashProductId(
```